### ISSUE TITLE
Lexer request: flex

### ISSUE THREAD

**Issue by seshaljain** [pygments/pygments#1893]
[flex](http://dinosaur.compilertools.net/) is a fast scanner generator which accepts regular expressions written in C. The C lexer (almost) works, but some escapes are marked as erroneous syntax.

**Comment by dman000** [comment]
Hello, I'm a student from the University of Adelaide. I’m working with a group of six students looking to contribute towards this project. We are interested in working on this issue if it is still available.

**Comment by Jean-Abou-Samra** [comment]
Sure. If nobody commented on the issue, it is still available. Feel free to take it.

**Comment by mattv52** [comment]
Hi, is there a way to have the lexer not start in the root state?
I would like our new lexer (inheriting from the existing C lexer) to start in a new state and then go to the root state later. Is this possible?
Thanks

**Comment by Jean-Abou-Samra** [comment]
I don't believe this exists, but there is a way around. Look at `SchemeLexer` and `LilyPondLexer` to see how this works. Basically, you can do

```
# C lexer:
tokens: {
    'root': [
        default('actual-root'),
    ],
    'actual-root': [
        # Fill in with what is currently in the 'root' state.
    ],
    # All instances of 'root' in the lexer should become 'actual-root'.
}

# Flex lexer

tokens: {
    'root': [
        # This is the start state of your new lexer.
    ],
    # To 'go to the root state', go to the 'actual-root' state.
}
```

That said, is Flex not grammar rules mixed with C code? Wouldn't `using()` be more appropriate for your purpose?
